examples/query/react/basic/src/store.ts

```diff
@@ -1,11 +1,28 @@
-import { configureStore } from '@reduxjs/toolkit'
+import {
+  combineReducers,
+  configureStore,
+  getDefaultMiddleware,
+} from '@reduxjs/toolkit'
 import { pokemonApi } from './services/pokemon'
 
-export const store = configureStore({
-  reducer: {
-    [pokemonApi.reducerPath]: pokemonApi.reducer,
-  },
-  // adding the api middleware enables caching, invalidation, polling and other features of `rtk-query`
-  middleware: (getDefaultMiddleware) =>
-    getDefaultMiddleware().concat(pokemonApi.middleware),
+const rootReducer = combineReducers({
+  [pokemonApi.reducerPath]: pokemonApi.reducer,
 })
+
+// We could do DeepPartial<RootState>? I don't know if preloadedState merges state or not
+export const setupStore = (initialState?: RootState) => {

```

Comment:
Can I just use `DeepPartial<RootState>` here? I I haven't had the chance to check if that will play nicely under the hoods with the `preloadedState` store parameter

Reply:
No, should be `PreloadedState<RootState>` as that will only allow full slices to be present or `undefined`, but not only half a slice being present (which is nothing `combineReducers` would account for)

Reply:
Sweet, doing that makes the `test-utils.tsx` setup easier and the typing for `IWithStoreOptions` more concise. Will fix in next commit.